Automattic/jetpack
# Google Fonts: Enqueue fonts used in block and Global Styles

#### Changes proposed in this Pull Request:

* Updates the Google Fonts module to be compatible with the most recent version of the WP_Webfonts API.
* Registered fonts are no longer enqueued by default, so we add hooks that enqueue Google Fonts used in block and Global Styles settings so those fonts load.
* This prevents dozens of font-face declarations from being printed out in the HTML source for unused fonts.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* On a site with the most recent version of Gutenberg active, enable the Google Fonts module (/wp-admin/admin.php?page=jetpack_modules).
* With a block theme enabled, to the Site editor.
* Go to Global styles and choose a Google Font from the Font family settings for all Text.
* Go to the Blocks settings in Global styles and choose a different Google Font for the font family setting for all blocks of that type, like Buttons (makes sure you have an instance of that block on the page).
* Select an individual block's settings and set yet another Google font for that individual block (like a List block... not all blocks have Font family settings, it partly depends on the theme).
* Now load the the page you were editing in the front end... make sure all of the fonts you selected are loaded. I do this by viewing the Network tab in browser tools and filtering for Fonts, or the filtering for the fonts.gstatic.com domain.
